src/screens/TransactionDetail.js

```diff
@@ -5,163 +5,98 @@
-      this.setState({
-        transaction: data.tx,
-        meta: data.meta,
-        spentOutputs: data.spent_outputs,
-        loaded: true,
-        success: true,
-      });
-    } else {
-      this.setState({ loaded: true, success: false, transaction: null });
-    }
-  }
+  const [transaction, setTransaction] = useState(null);
+  /* meta {Object} Metadata of loaded transaction received from the server */
+  const [meta, setMeta] = useState(null);
+  /* loaded {boolean} If had success loading transaction from the server */
+  const [loaded, setLoaded] = useState(false);
+  /* spentOutputs {Object} Spent outputs of loaded transaction received from the server */
+  const [spentOutputs, setSpentOutputs] = useState(null);
+  /* confirmationData {Object} Confirmation data of loaded transaction received from the server */
+  const [confirmationData, setConfirmationData] = useState(null);
 
   /**
    * Get transaction in the server
    */
-  updateTxInfo = id => {
-    txApi.getTransaction(id).then(
-      data => {
-        this.txReceived(data);
-        if (data.success && !hathorLib.transactionUtils.isBlock(data.tx)) {
-          this.getConfirmationData();
-        }
-      },
-      e => {
-        // Error in request
-        console.log(e);
-      }
-    );
-  };
+  const updateTxInfo = useCallback(async id => {
+    const txData = await txApi.getTransaction(id);
 
-  /**
-   * Get transaction metadata from explorer service
-   */
-  updateTxMetadata = id => {
-    metadataApi.getDagMetadata(id).then(data => {
-      if (data) {
-        this.setState(oldState => {
-          return {
-            transaction: {
-              ...oldState.transaction,
-              meta: data,
-            },
-          };
-        });
-      }
-    });
-  };
+    setLoaded(true);
+    if (!txData.success) {
+      setTransaction(null);
+      setMeta(null);
+      setSpentOutputs(null);
+      setConfirmationData(null);
+      return;
+    }
 
-  /**
-   * Update transaction information when loading page
-   */
-  getTx() {
-    this.updateTxInfo(this.props.match.params.id);
-    this.updateTxMetadata(this.props.match.params.id);
-  }
+    // Update state after receiving the transaction response back from the server
+    setTransaction(txData.tx);
+    setMeta(txData.meta);
+    setSpentOutputs(txData.spent_outputs);
 
-  /**
-   * When transaction changed in the page we need to load the new one and the new confirmation data
-   */
-  componentDidUpdate(prevProps, prevState, snapshot) {
-    if (this.props.match.params.id !== prevProps.match.params.id) {
-      this.getTx();
+    // Get accumulated weight and confirmation level of the transaction
+    if (!hathorLib.transactionUtils.isBlock(txData.tx)) {
+      const confirmationDataResponse = await txApi.getConfirmationData(id);
+      setConfirmationData(confirmationDataResponse);
+    }
+
+    // Get transaction metadata from explorer service, overwriting the one already obtained
```

Comment:
Note: This logic of overwriting the `meta` state was already present, I just made it more obvious by removing the intermediate functions that were doing the double work.

This may be fixed or better structured at a future moment, but the scope of this PR is only to refactor and not to change the logic.